doc/guides/gcov.rst

```diff
@@ -0,0 +1,44 @@
+.. _gcov:
+
+Coverage reports via GCOV
+#########################
+
+Overview
+********
+This feature will allow the extraction of the coverage reports via GCC GCOV tool.
```

Comment:
Could use some more introduction, and a rewrite to get rid of all the *Therefore's*:
```
`gcov<https://gcc.gnu.org/onlinedocs/gcc/Gcov.html>`_ is a test
coverage program used together with the GCC compiler to analyze
and create test coverage reports for your programs, helping you
create more efficient, faster running code and discovering untested
code paths. 

In Zephyr, gcov collects coverage profiling data in RAM (and not
to a file system) while your application is running. Support for gcov
collection and reporting is limited by available RAM size and so
is currently enabled only for QEMU emulation of embedded targets.
```